File: number-guessing-game/guessing_game.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    srand(time(0));
    int secret = rand() % 100 + 1;
    int guess, attempts = 0;

    cout << "===== Number Guessing Game =====" << endl;
    cout << "I picked a number between 1 and 100!" << endl;

    while (true) {
        cout << "Your guess: ";
        cin >> guess;
        attempts++;

        if (guess < secret)
            cout << "Too low! Try again." << endl;
        else if (guess > secret)
            cout << "Too high! Try again." << endl;
        else {
            cout << "Correct! You got it in " << attempts << " attempts!" << endl;
            break;
        }
    }

    return 0;
}

File: number-guessing-game/guessing_game.py
import random

def guessing_game():
    print("===== Number Guessing Game =====")
    
    secret = random.randint(1, 100)
    attempts = 0
    
    print("I picked a number between 1 and 100!")
    
    while True:
        guess = int(input("Your guess: "))
        attempts += 1
        
        if guess < secret:
            print("Higher number please!")
        elif guess > secret:
            print("Lower number please!")
        else:
            print(f"Yes! You got it in {attempts} attempts!")
            break

guessing_game()
